### DISCUSSION TITLE
How should the product of inertia be computed?

### DISCUSSION THREAD

**Discussion by aarondewindt** [JSBSim-Team/jsbsim#442] [Q&A]
On some sources such as "Aircraft Control and Simulation: Dynamics, Controls Design, and Autonomous Systems" by "Brian L. Stevens", the product moment of inertia are computed with.

$$I_{xz}=\int xzdm$$

However on some other sources the product of inertia is computed the equation shown below. So this would be the value in the inertia tensor.

$$I_{xz}=-\int xzdm$$

After looking through the JSBSim source code we think we should fill in values calculated by the second equation. Is this the case?

**Comment by seanmcleod** [comment]
See the following:

https://github.com/JSBSim-Team/jsbsim/blob/a35be5ce6f7d7ca0ef9a11a62563e54d769bc4f4/src/models/FGMassBalance.h#L68-L71

Which was updated as part of the following commit - https://github.com/JSBSim-Team/jsbsim/commit/03763374fcd688bdcf9faf6164106a68f8ba3c0b#diff-ef28318d289aee0dd2e968ada735838f3e7ad8daf7da1eaea2448ed27ec56b4a

With further info in this earlier issue - https://github.com/JSBSim-Team/jsbsim/issues/125

**Comment by aarondewindt** [comment]
@seanmcleod Thanks!!

**Comment by Jemonde** [comment]
Hi Sean ! 
good to see that you are on every touchy subjects.. 

I wonder.. why is that ? The change of sign won't affect the behavior of the plane ? Is it a convention ? .. 
When I extract the Inertial matrix from my CAD at the CoG but in a different frame (CAD frame) and when I change the frame of expression of this matrix, I only get positives values for the cross products in the structural frame... 

So you tell me I should change the sign of the cross products? 

Thank you in advance for your time,

**Comment by bcoconni** [comment]
> The change of sign won't affect the behavior of the plane ? Is it a convention ? ..

@Jemonde Changing the sign of cross product inertia does affect the model behavior. However JSBSim allows both sign convention to be used (internally changing the sign if the input sign convention does not match JSBSim internal sign convention).
https://github.com/JSBSim-Team/jsbsim/blob/5ad2694cb75be6900ee0d9ca8785b247d7670162/src/models/FGMassBalance.cpp#L118-L127
You might find interesting reading the discussions in the PR #502 where all the details have been discussed.
